forum/qa-content/qa-page.js

```diff
@@ -582,192 +582,100 @@ function qa_ajax_error()
      * Feature: Post content preview as Modal
      * Date: 07.07.2016r.
      */
-    function postPreview(ckeCurrentInstance, placeForBtn)
+
+    function destroyModal()
     {
-        var modalParent = document.querySelector('.qa-main-wrapper');
+        const modal = document.querySelector('.post-preview-parent');
+        modal.parentNode.removeChild(modal);
 
-        var modalBackground = document.createElement('div');
-        modalBackground.classList.add('modal-background');
+        const modalBackground = document.querySelector('.modal-background');
+        modalBackground.parentNode.removeChild(modalBackground);
+    }
 
+    function createPostPreviewModal(ckeInstanceName)
+    {
+        const modal = document.createElement('div');
+        modal.classList.add('post-preview-parent');
 
-        var showModalBtn = document.createElement('input');
-        showModalBtn.type = 'button';
-        showModalBtn.id = 'get-content-preview';
-        showModalBtn.value = 'Podgląd posta';
-        showModalBtn.classList.add('qa-form-tall-button', 'get-content-preview');
+        const closeModalButton = document.createElement('button');
+        closeModalButton.innerHTML = 'X';
+        closeModalButton.classList.add('close-preview-btn');
+        closeModalButton.addEventListener('click', destroyModal);
+        modal.appendChild(closeModalButton);
 
-        if (!placeForBtn)
-        {
-            var alternativePlaceForBtn = document.querySelector('.qa-form-tall-buttons [value="Zadaj pytanie"]')
-                                        || document.querySelector('.qa-form-tall-buttons [value="Zapisz"]')
-                                        || document.querySelector('.qa-form-tall-buttons [value="Odpowiedz"]');
-            alternativePlaceForBtn.parentNode.appendChild(showModalBtn);
-        } else if ( placeForBtn.querySelector( '#' + showModalBtn.id ) )
-            return;
-        else if (placeForBtn)
-            placeForBtn.appendChild(showModalBtn);
-
-        function modalEventHandler(modalWrapper, closeBtn)
-        {
-            function hideModal()
-            {
-                var modalWrapperParent = modalWrapper.parentNode;
+        const modalContent = document.createElement('div');
+        modalContent.innerHTML = CKEDITOR.instances[ckeInstanceName].getData();

```

Comment:
j.w.: `innerHTML` -> `textContent` (jeśli zapisujesz tylko tekst)`

Reply:
Z tego co się orientuję, to w treści posta znajdują się dane w HTML - w ten sposób jest realizowane pogrubienie, zmiana kolorów, bloczki kodu… więc tutaj chyba trzeba zostać przy innerHTML.